python_modules/dagster-framework/snowflake/dagster_framework/snowflake/solids.py

```diff
@@ -0,0 +1,187 @@
+
+        description = check.opt_str_param(
+            description,
+            'description',
+            'This solid is a generic representation of a parameterized Snowflake query.',
+        )
+
+        description = (
+            description
+            or 'This solid is a generic representation of a parameterized Snowflake query job.'
+        )
+
+        def _define_snowflake_transform_fn(context, _):  # pylint: disable=too-many-locals
+            '''Define Snowflake execution.
+
+            This function defines how we'll execute the Snowflake SQL query.
+            '''
+            # Extract parameters from config
+            (
+                account,
+                user,
+                password,
+                database,
+                schema,
+                role,
+                warehouse,
+                autocommit,
+                client_prefetch_threads,
+                client_session_keep_alive,
+                login_timeout,
+                network_timeout,
+                ocsp_response_cache_filename,
+                validate_default_parameters,
+                paramstyle,
+                timezone,
+            ) = [
+                context.solid_config.get(k)
+                for k in (
+                    'account',
+                    'user',
+                    'password',
+                    'database',
+                    'schema',
+                    'role',
+                    'warehouse',
+                    'autocommit',
+                    'client_prefetch_threads',
+                    'client_session_keep_alive',
+                    'login_timeout',
+                    'network_timeout',
+                    'ocsp_response_cache_filename',
+                    'validate_default_parameters',
+                    'paramstyle',
+                    'timezone',
+                )
+            ]
+
+            conn_args = {
+                'user': user,
+                'password': password,
+                'account': account,
+                'schema': schema,
+                'database': database,
+                'role': role,
+                'warehouse': warehouse,
+                'autocommit': autocommit,
+                'client_prefetch_threads': client_prefetch_threads,
+                'client_session_keep_alive': client_session_keep_alive,
+                'login_timeout': login_timeout,
+                'network_timeout': network_timeout,
+                'ocsp_response_cache_filename': ocsp_response_cache_filename,
+                'validate_default_parameters': validate_default_parameters,
+                'paramstyle': paramstyle,
+                'timezone': timezone,
+            }
+
+            # We can't pass None values to snowflake.connector.connect() because they will override
+            # the default values set within the connector; remove them from the conn_args dict
+            conn_args = {k: v for k, v in conn_args.items() if v}
```

Comment:
`if v is not None` -- otherwise this will also filter out `False`, `''`, `[]`, etc.

Reply:
damn. yes. I'll do a follow-up PR with tests, we should have tests around all of this code